ChildProject/annotations.py

```diff
@@ -0,0 +1,218 @@
+        'OC0': 'OCH',
+        'UC1': 'OCH',
+        'UC2': 'OCH',
+        'UC3': 'OCH',
+        'UC4': 'OCH',
+        'UC5': 'OCH',
+        'UC6': 'OCH'
+    }
+
+
+    def __init__(self, project):
+        self.project = project
+        self.annotations = None
+        self.errors = []
+
+        if not isinstance(project, ChildProject):
+            raise ValueError('project should derive from ChildProject')
+
+        project.read()
+
+        index_path = os.path.join(self.project.path, 'annotations/annotations.csv')
+        if not os.path.exists(index_path):
+            open(index_path, 'w+').write(','.join([c.name for c in self.INDEX_COLUMNS]))
+
+        errors, warnings = self.read()
+
+    def read(self):
+        table = IndexTable('input', path = os.path.join(self.project.path, 'annotations/annotations.csv'), columns = self.INDEX_COLUMNS)
+        self.annotations = table.read()
+        errors, warnings = table.validate()
+        return errors, warnings
+
+    def validate(self):
+        errors, warnings = [], []
+
+        for annotation in self.annotations.to_dict(orient = 'records'):
+            segments = IndexTable(
+                'segments',
+                path = os.path.join(self.project.path, 'annotations', annotation['annotation_filename']),
+                columns = AnnotationManager.SEGMENTS_COLUMNS
+            )
+
+            segments.read()
+            res = segments.validate()
+            errors += res[0]
+            warnings += res[1]
+
+        return errors, warnings
+        
+
+    def load_textgrid(self, filename):
+        path = os.path.join(self.project.path, 'raw_annotations', filename)
+        textgrid = pympi.Praat.TextGrid(path)
+
+        segments = []
+        for tier in textgrid.tiers:
+            for interval in tier.intervals:
+                tier_name = tier.name.strip()
+
+                if tier_name == 'Autre':
+                    continue
+
+                if interval[2] == "":
+                    continue
+
+                segment = {
+                    'annotation_file': filename,
+                    'segment_onset': "{:.3f}".format(interval[0]),
+                    'segment_offset': "{:.3f}".format(interval[1]),
+                    'speaker_id': tier_name,
+                    'ling_type': interval[2] if interval[2] else "",
+                    'speaker_type': self.SPEAKER_ID_TO_TYPE[tier_name] if tier_name in self.SPEAKER_ID_TO_TYPE else 'NA',
+                    'vcm_type': 'NA',
+                    'lex_type': 'NA',
+                    'mwu_type': 'NA',
+                    'addresseee': 'NA',
+                    'transcription': 'NA'
+                }
+
+                segments.append(segment)
+
+        return pd.DataFrame(segments)
+
+    def load_eaf(self, filename):
+        path = os.path.join(self.project.path, 'raw_annotations', filename)
+        eaf = pympi.Elan.Eaf(path)
+
+
+        return None
+
+    def load_vtc_rttm(self, filename):
```

Comment:
is the logic correct here ?
shouldn't CHI be translated into OCHI and KCHI in CHI or something of this sort ?

Reply:
the logic is perfect.
You're right, Marvin's VTC uses KCHI for key child and CHI for other children (confusingly, since everyone else in the language development world uses CHI for the key child). So yes, you'd need to add to the section starting line 37, with the dictionary, a translation CHI -> OCH (notice no I); and KCHI -> CHI